### PR TITLE
Kick-off Jenkinsfile improvements

### PR DESCRIPTION
**Pull request by juliangieseke**
This PR refactors our CI files to allow us to move forward. Need more work :(

### CODE REVIEW

**Comment by ahoskins** [comment]
So this PR will have everything in our new release process and always-open latest PR? We can cherry-pick the changes for updating github into here once they are "stable". 😅

**Comment by juliangieseke** [comment]
Yes, but I'm holding this back until we got time to sync about it. dont want to block you guys too much 👍

File: ci/update-latest [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+echo 'create tar'
+
+echo 'upload as latest'
+aws s3 cp 'release.tar.gz' s3://$AWS_BUCKET/$BUCKET_PATH/$RELEASE_NAME
```
Comment (juliangieseke) [inline ci/update-latest]: this should have been in the next commit :(

**Comment by juliangieseke** [comment]
sorry for this being one giant commit, I f*ed up the rebase :(

**Comment by juliangieseke** [comment]
closing this in favor of #2796 - wanted to refactor a bit to make the merge easier, but wasnt able to finish it in time.

**Pull request closed without merging**